pkg/integrations/v4/config/loader.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"errors"
+	"github.com/newrelic/infrastructure-agent/internal/integrations/v4/files"
+	"github.com/newrelic/infrastructure-agent/pkg/config/envvar"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+var clLog = log.WithComponent("integrations.config.Loader")
+
+// LegacyYAML is not an actual error. Used for discarding V3 plugins
+var LegacyYAML = errors.New("file format belongs to the old integrations format")
+
+const (
+	// Integrations V3 configs use the "instances" key word
+	// In the current Integrations config, we use "integrations"
+	// The two fields below will allows us distinguish between both, to ignore V4
+	integrationsField    = "integrations"
+	LegacyInstancesField = "instances"
+)
+
+// Loader will read and parse integrations v4 config files.
+type Loader interface {
+	// Load reads all the configuration files in a given directory. If path is a file instead of a directory
+	// will try to read it as a single configuration file.
+	Load(path string) (YAMLMap, error)
+
+	// LoadFile reads the configuration file.
+	LoadFile(file string) (YAML, error)
+}
+
+type pathLoader struct {
+	isDirectory func(path string) (bool, error)
+}
+
+// NewPathLoader returns a new instance of a config Loader.
+func NewPathLoader() Loader {
+	return &pathLoader{
+		isDirectory: isDirectory,
+	}
+}
+
+// Load reads all the configuration files in a given directory. If path is a file instead of a directory
+// will try to read it as a single configuration file.
+func (pl *pathLoader) Load(path string) (YAMLMap, error) {
+	isDir, err := isDirectory(path)
```

Comment:
I think that this should be `pl.isDirectory` shouldn't it? ☝️

Reply:
you're right 👍